### PR TITLE
Fix ghost avatars/displaynames

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8570fe0c19d444664d50fbfce7e58ba07b03a2b3 available at: https://gist.github.com/27fc859b36f338a6b9be<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5093.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5093.dmg

File: interface/src/avatar/AvatarManager.cpp [hunk 1]
```diff
@@ -171,10 +175,6 @@ void AvatarManager::removeAvatar(const QUuid& sessionUUID) {
             _avatarFades.push_back(avatarIterator.value());
             _avatarHash.erase(avatarIterator);
         }
-        render::ScenePointer scene = Application::getInstance()->getMain3DScene();
-        render::PendingChanges pendingChanges;
-        avatar->removeFromScene(avatar, scene, pendingChanges);
```
Comment (ZappoMan) [inline interface/src/avatar/AvatarManager.cpp]: if you delete these lines... when it is being removed?

Comment (Atlante45) [inline interface/src/avatar/AvatarManager.cpp]: when we remove it from the fades.

File: interface/src/avatar/AvatarManager.cpp [hunk 2]
```diff
@@ -121,26 +121,30 @@ void AvatarManager::simulateAvatarFades(float deltaTime) {
...
-
+    
+    render::ScenePointer scene = Application::getInstance()->getMain3DScene();
+    render::PendingChanges pendingChanges;
     while (fadingIterator != _avatarFades.end()) {
-        Avatar* avatar = static_cast<Avatar*>(fadingIterator->get());
+        auto avatar = std::static_pointer_cast<Avatar>(*fadingIterator);
         avatar->setTargetScale(avatar->getScale() * SHRINK_RATE, true);
         if (avatar->getTargetScale() < MIN_FADE_SCALE) {
+            avatar->removeFromScene(*fadingIterator, scene, pendingChanges);
```
Comment (Atlante45) [inline interface/src/avatar/AvatarManager.cpp]: @ZappoMan here

**Pull request merged**